service/src/test/java/bio/terra/workspace/db/DbRetryTest.java

```diff
@@ -0,0 +1,37 @@
+package bio.terra.workspace.db;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import bio.terra.common.retry.transaction.TransactionRetryProperties;
+import bio.terra.workspace.common.BaseUnitTest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.CannotSerializeTransactionException;
+
+public class DbRetryTest extends BaseUnitTest {
+
+  @Autowired FakeDao fakeDao;
+  @Autowired TransactionRetryProperties retryProperties;
+
+  private static final CannotSerializeTransactionException RETRY_EXCEPTION =
+      new CannotSerializeTransactionException("test");
+
+  @BeforeEach
+  public void setup() {
+    fakeDao.reset();
+  }
+
```

Comment:
Maybe add test cases for InterruptedException

Reply:
If the thread is killed, there will be an InterruptedException delivered at some level. Maybe it will show up inside the database client waiting for a postgres response.

I was suggesting throwing one. In theory, the retryable code will catch it and return a BackOffInterruptedException. Right?